Sources/_StringProcessing/Algorithms/Matching/MatchReplace.swift

```diff
@@ -110,4 +110,82 @@ extension RangeReplaceableCollection where SubSequence == Substring {
       with: replacement,
       maxReplacements: maxReplacements)
   }
+
+  /// Returns a new collection in which all occurrences of a sequence matching
+  /// the given regex are replaced by another regex match.
+  /// - Parameters:
+  ///   - regex: A regex describing the sequence to replace.
+  ///   - replacement: A closure that receives the full match information,
+  ///   including captures, and returns a replacement collection.
+  ///   - subrange: The range in the collection in which to search for `regex`.
+  ///   - maxReplacements: A number specifying how many occurrences of the
+  ///   sequence matching `regex` to replace. Default is `Int.max`.
+  /// - Returns: A new collection in which all occurrences of subsequence
+  /// matching `regex` are replaced by `replacement`.
+  public func replacing<R: RegexComponent, Replacement: Collection>(
+    _ regex: R,
+    with replacement: (Regex<R.Output>.Match) throws -> Replacement,
```

Comment:
@milseman this would replace `_MatchResult` boilerplate

Reply:
That's great!